e2e/snapshot_ops_test.go

```diff
@@ -343,3 +337,33 @@ func testSnapshotMount(t *testing.T, tc *testCluster) {
 	err = umntCmd.Run()
 	r.Nil(err, fmt.Sprintf("unmount failed: %s", err))
 }
+
+func testSnapshotValidation(t *testing.T) {
+	r := require.New(t)
+
+	snapshotCreateReq := api.SnapCreateReq{
+		VolName:   clonename,
+		SnapName:  snapname,
+		TimeStamp: true,
+	}
+	_, err := client.SnapshotCreate(snapshotCreateReq)
+	r.Nil(err, "snapshot create failed")
+
+	r.Nil(client.VolumeStop(clonename), "volume stop failed")
+
+	if err := client.VolumeDelete(clonename); err == nil {
```

Comment:
question:
am bit confused here, can you explain what you are trying to achieve

if you want to fail the text case if volume delete succeed, can we do
```
r.NotNil(err,"Volume delete succeeded when the snapshot is existing for the volume")
```

Reply:
If a snapshot exists for a volume then delete should fail. So I'm trying to delete a volume which has a snapshot.